assets/js/theme/common/product-details.js

```diff
@@ -108,15 +109,22 @@ export default class ProductDetails {
             if (type === 'set-rectangle' || type === 'set-radio' || type === 'swatch' || type === 'input-checkbox' || type === 'product-list') {
                 const checked = value.querySelector(':checked');
                 if (checked) {
+                    const getSelectedOptionLabel = () => {
+                        const productVariantslist = convertIntoArray(value.children);
+                        return productVariantslist.reduce((res, field) => {
+                            if (field.dataset.productAttributeValue === checked.value) res = field; // eslint-disable-line no-param-reassign
+                            return res;
+                        });
```

Comment:
I suggest to use .find instead of .reduce 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Reply:
this code is aimed to run only in IE and use a method that requires a polyfill is not 'tasty')